solarsystemoptional.rb

```diff
@@ -0,0 +1,187 @@
+  # return pdata
+  pdata = "Planet: #{@planet_name}\n" +
+  ".  ✧　 * ✫  ⊹ A year here takes #{@year_length} days.\n" +".  ✧　 * ✫  ⊹ I am #{@distance_from_the_sun} million miles from the sun.\n" + ".  ✧　 * ✫  ⊹ My sailor scout is #{scout_name}.\n" + ".  ✧　 * ✫  ⊹ She is the #{@soldier_of}." + "\n "
+  return pdata
+end
+
+end # End of planet class
+
+# Creates 5 planets with 5 attributes
+mercury = Planet.new(
+   {
+    planet_name: "mercury",
+    year_length: 88,
+    distance_from_the_sun: 36,
+    scout_name: "Ami Mizuno",
+    soldier: "Soldier of Water and Wisdom"
+  })
+
+venus = Planet.new (
+{
+    planet_name: "venus",
+    year_length: 226,
+    distance_from_the_sun: 64,
+    scout_name: "Minako Aino",
+    soldier: "Soldier of Love and Beauty"
+  })
+
+mars = Planet.new (
+  {
+    planet_name: "mars",
+    year_length: 687,
+    distance_from_the_sun: 141,
+    scout_name: "Rei Hino",
+    soldier: "Soldier of Fire and Passion"
+  })
+
+  jupiter = Planet.new (
+  {
+    planet_name: "jupiter",
+    year_length: 4380,
+    distance_from_the_sun: 484,
+    scout_name: "Makoto Kino",
+    soldier: "Soldier of Thunder and Courage"
+  })
+
+  # Creates a new solar system with an array of the 5 planets
+  senshi = SolarSystem.new ([mercury, venus, mars, jupiter])
+
+  # # Prints a list of planets
+  # puts senshi.planets
+  #
+  # # Prints information on the planet mercury
+  # puts mercury.planet_data
+
+  saturn = Planet.new(
+
+  {
+      planet_name: "saturn",
+      year_length: 226,
+      distance_from_the_sun: 64,
+      scout_name: "Hotaru Tomoe",
+      soldier: "Soldier of Destruction and Silence"
+    })
+
+senshi.new_planet(saturn)
+
+
+puts "\n.  ✧　 * ✫  ⊹ Welcome to the Sailor Senshi System.  ✧　 * ✫  ⊹ "
+puts ".  ✧　 * ✫  ⊹ Please type in a planet to learn more ✧　 * ✫  ⊹ "
+puts ".  ✧　 * ✫  ⊹ Or type NEW to make one of your own!.  ✧　 * ✫  ⊹ \n"
+puts senshi.planets
+print ".  ✧　 * ✫  ⊹ "
+
+user_input = gets.chomp
+
+# Gets input from user until they enter stop
+# If they enter new, creates a new planet based on user input and adds it to the system
+# If they enter name of planet, it will print data
+until user_input == "stop"
+  if "#{user_input}" == "new"
+    print "\n✫new planet name: "
+    new_name = gets.chomp
+    print "\n*year length in days: "
+    new_year = gets.chomp.to_i
+    print "\n*distance from sun in millions of miles: "
+    new_distance = gets.chomp.to_i
+    print "\n⊹scout name: "
+    new_scout = gets.chomp
+    print "\n✫soldier of: "
+    new_soldier = "Soldier of " + gets.chomp
+
+    user_planet = Planet.new (
+    {
+      planet_name: "#{new_name}",
+      year_length: new_year,
+      distance_from_the_sun: new_distance,
+      scout_name: "#{new_scout}",
+      soldier: "#{new_soldier}"
+    })
+    senshi.new_planet(user_planet)
+    print "\nthanks! user planet is saved.\n"
+  elsif
```

Comment:
this is an ambiguous piece of code: What is `elsif` evaluating against? The block of code in the each loop? Try to make it more clear next time